src/main/java/com/solace/samples/java/patterns/DirectReplierBlocking.java

```diff
@@ -0,0 +1,156 @@
+package com.solace.samples.java.patterns;
+
+import com.solace.messaging.MessagingService;
+import com.solace.messaging.config.SolaceProperties;
+import com.solace.messaging.config.profile.ConfigurationProfile;
+import com.solace.messaging.publisher.OutboundMessage;
+import com.solace.messaging.publisher.OutboundMessageBuilder;
+import com.solace.messaging.receiver.InboundMessage;
+import com.solace.messaging.receiver.RequestReplyMessageReceiver;
+import com.solace.messaging.resources.TopicSubscription;
+
+import java.io.IOException;
+import java.util.Properties;
+
+/**
+ * This class demonstrates the usage of the Solace Java API to create a Replier class.
+ * This implementation focuses on the blocking behaviour of the API.
+ * The mechanism of the Request-Reply pattern is defined in more detail over here : <a href="https://tutorials.solace.dev/jcsmp/request-reply/">Solace Request/Reply pattern</a>
+ * <p>
+ * Refer to the DirectRequesterBlocking class for the request component of the flow.
+ */
+public class DirectReplierBlocking {
+
+    private static final String SAMPLE_NAME = DirectReplierBlocking.class.getSimpleName();
+    private static final String TOPIC_PREFIX = "solace/samples/";  // used as the topic "root"
+    private static final String API = "Java";
+    private static volatile int msgRecvCounter = 0;                   // num messages sent
+    private static volatile boolean hasDetectedDiscard = false;  // detected any discards yet?
+    private static volatile boolean isShutdown = false;          // are we done yet?
+
+    public static void main(String... args) throws IOException {
+
+        //1. Make sure that you have all the connection parameters.
+        validateStartupParameters(args);
+
+        System.out.println(API + " " + SAMPLE_NAME + " initializing...");
+
+        //2. Set up the properties including username, password, vpnHostUrl and other control parameters.
+        final Properties properties = new Properties();
+        setupPropertiesForConnection(properties, args);
+
+        //3. Create the MessagingService object and establishes the connection with the Solace event broker
+        final MessagingService messagingService = com.solace.messaging.MessagingService.builder(ConfigurationProfile.V1)
+                .fromProperties(properties)
+                .build();
+        messagingService.connect();  // blocking connect to the broker
+
+        //4. Register event handlers and callbacks for connection error handling.
+        setupConnectivityHandlingInMessagingService(messagingService);
+
+        //5. Build and start the Receiver object
+        RequestReplyMessageReceiver requestReplyMessageReceiver = messagingService.requestReply().
+                createRequestReplyMessageReceiverBuilder().build(TopicSubscription.of(TOPIC_PREFIX + "*/direct/>"));
```

Comment:
We should add `request` in the topic so this replier only tries to respond to requests. Looks like @TamimiGitHub used `solace/samples/python/direct/request/>` for the Python replier.